### ISSUE TITLE
Refactor into PyTorch Lightning Modules

### ISSUE THREAD

**Issue by andrewjong** [andrewjong/ShineOn-Virtual-Tryon#36]
Because we will now have several models to test (GMM, TOM, SAMS), we should transition to PT Lightning such that everything is modular. We want each model to handle its own input, loss, and optimization; which PT Lightning easily does! Otherwise we'd messily have duplicate train functions littered throughout train.py.

PT Lightning also makes it easy to [wrap a pretrained model](https://pytorch-lightning.readthedocs.io/en/0.7.0/transfer_learning.html) (i.e. GMM) within another model.

Doing this refactoring will make it easy to integrate the planned Self-Attentive Multi-Spade (SAMS) model.
